app/src/main/java/org/projectbuendia/client/user/UserStore.java

```diff
@@ -45,90 +46,61 @@
-        try {
-            client = App.getInstance().getContentResolver()
-                .acquireContentProviderClient(Users.CONTENT_URI);
-
-            // Request users from database.
-            try {
-                cursor = client.query(Users.CONTENT_URI, null, null, null, Users.FULL_NAME);
-            } catch (RemoteException e) {
-                LOG.e(e, "Error retrieving users from database");
-            }
-
-            // If no data was retrieved from database, force a sync from server.
-            if (cursor == null || cursor.getCount() == 0) {
-                LOG.i("Database contains 0 users; fetching from server");
-                return syncKnownUsers();
-            }
-            LOG.i("Found " + cursor.getCount() + " users in db");
-
-            // Initiate users from database data and return the result.
-            int fullNameColumn = cursor.getColumnIndex(Users.FULL_NAME);
-            int uuidColumn = cursor.getColumnIndex(Users.UUID);
-            Set<JsonUser> result = new HashSet<>();
-            while (cursor.moveToNext()) {
-                JsonUser user =
-                    new JsonUser(cursor.getString(uuidColumn), cursor.getString(fullNameColumn));
-                result.add(user);
-            }
-            return result;
-        } catch (SQLiteException e) {
-            LOG.w(e, "Error retrieving users from database; fetching from server");
-            return syncKnownUsers();
-        } finally {
-            if (cursor != null) {
-                cursor.close();
-            }
-            if (client != null) {
-                client.release();
-            }
+        Set<JsonUser> users = getUsersFromDb();
+        if(users.isEmpty()) {
+            LOG.i("Database contains no user; fetching from server");
+            users = syncKnownUsers();
         }
+
+        LOG.i(String.format("Found %d users in db", users.size()));
+        return users;
     }
 
     /** Syncs known users with the server. */
     public Set<JsonUser> syncKnownUsers()
         throws ExecutionException, InterruptedException, RemoteException,
         OperationApplicationException {
-        RequestFuture<List<JsonUser>> future = RequestFuture.newFuture();
-        App.getServer().listUsers(null, future, future);
-        List<JsonUser> users = future.get();
-        HashSet<JsonUser> userSet = new HashSet<>();
-        userSet.addAll(users);
+        Set<JsonUser> users = getUsersFromServer();
+        updateDatabase(users);
+        return users;
+    }
+
+    /** Adds a new user, both locally and on the server. */
+    public JsonUser addUser(JsonNewUser user) throws VolleyError {
+        JsonUser newUser = addUserOnServer(user);
+        addUserLocally(newUser);
+        return newUser;
+    }
 
-        LOG.i("Got %d users from server; updating local database", users.size());
+    /** Deletes a user, both locally and on the server. */
+    public JsonUser deleteUser(JsonUser user) {
```

Comment:
Thanks! Ok to merge.

In general, it's better to delete unused code because it makes the process of changing code in the future simpler.